## Bonus:  Other Recommended Analyses

### Temperature Analysis I
* Is there a meaningful difference between the temperature in June and December

In [1]:
# Import dependencies
import pandas as pd
from datetime import datetime as dt
from scipy import stats

In [2]:
# Read csv file into dataframe
# "tobs" is "temperature observations"
df = pd.read_csv('hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Check data types
df.date.dtype

dtype('O')

In [4]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [5]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])
df

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73
...                 ...        ...   ...   ...
2017-08-19  USC00516128 2017-08-19  0.09    71
2017-08-20  USC00516128 2017-08-20   NaN    78
2017-08-21  USC00516128 2017-08-21  0.56    76
2017-08-22  USC00516128 2017-08-22  0.50    76
2017-08-23  USC00516128 2017-08-23  0.45    76

[19550 rows x 4 columns]

In [6]:
# Drop the date column
df = df.drop(columns='date')
df

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73
...                 ...   ...   ...
2017-08-19  USC00516128  0.09    71
2017-08-20  USC00516128   NaN    78
2017-08-21  USC00516128  0.56    76
2017-08-22  USC00516128  0.50    76
2017-08-23  USC00516128  0.45    76

[19550 rows x 3 columns]

### Compare June and December Dates

In [7]:
# Filter data for June and December
jun_data = df[df.index.month == 6]
dec_data = df[df.index.month == 12]

In [8]:
# Calculate the June mean
jun_data.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [9]:
# Calculate the December mean
dec_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [10]:
# Create collections of temperature data
jun_temp = jun_data.tobs
dec_temp = dec_data.tobs

In [11]:
# Run paired t-test; t-test compares two means
stats.ttest_ind(jun_temp, dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

##  Analysis Result
* In June and December, the mean temperatures only differed by 3.9 degrees.  The t-test p-value result was 3.9025129038616655e-191.  This low p-value indicates there is no statistical significant difference.